person.rb

```diff
@@ -13,12 +16,51 @@ def can_use_services?
     of_age? or @parent_permission
   end
 
+  def correct_name
+    @name
+  end
+
   private
 
   def of_age?
     age >= 18
   end
 end
+# -----------------
+
+class Decorator < Nameable
+  attr_accessor :nameable
+
+  # @param [nameable] nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  # The Decorator delegates all work to the wrapped nameable.
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+# Concrete Decorators call the wrapped object and alter its result in some way.
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+# Decorators can execute their behavior either before or after the call to a
+# wrapped object.
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name[0..8]
+  end
+end
```

Comment:
- [ ] As per the project's requirement, `@nameable.correct_name` has a maximum of 10 characters. If it's longer it should trim the word. Currently, it's taking 9 characters. Kindly fix it. :+1: